### ISSUE TITLE
There's no way to override `@pjnew` values in derived classes

### ISSUE THREAD

**Issue by duckpuppy** [projen/projen#3768]
My company is extending a few of the base Projen project types so we can adjust workflows and add things like internal Artifactory authentication support without requiring our developers to copy boilerplate config snippets everywhere. Our first extension was to the jsii-project type as that would form the base for all of our other custom project types. One thing we want to do is pin the `jsiiVersion` attribute to `~5.3.0`. However, `projen new` creates a `.projenrc.ts` with `jsiiVersion: "~5.4.0"` because of the `@pjnew` jsdoc tag on the base class, and there seems to be no way to override that. We have found that we are able to force 5.3.0 in our derived class just by hard-coding that value no matter what's in the RC file, but having that line in there creates more confusion for teams using our new base class.

My question is just that: Does someone have a method for overriding that line in the generated `.projenrc.ts` file when creating a new project from our derived project type, either by eliminating that line entirely or providing a different value somehow?

If not, it would be nice to have a way to do that.

**Comment by mrgrain** [comment]
Possibly. Do you have a repro or minimal example?

What does the generated `.jsii` assembly look like?

**Comment by duckpuppy** [comment]
The repro is really just `projen new jsii` and seeing the line `jsiiVersion: "~5.4.0"` in the generated `.projenrc.ts` file. Any class that extends the core JSII project class will also exhibit that behavior. Because that value is coming from a doc tag and the field can't be overridden in a derived class to add a different `@pjnew` value (the JSII compiler won't allow it due to issues with C#, apparently), there seems to be no way currently to modify the values placed into the projen config file when creating a new project. I was thinking of project types having a way to hook into config file generation (a exported function that's called by `projen new` when creating a new app from that project type maybe?)

**Comment by mrgrain** [comment]
> The repro is really just `projen new jsii` and seeing the line `jsiiVersion: "~5.4.0"` in the generated `.projenrc.ts` file. 

I'm sure you meant something else, but this behavior is obviously intended.

You can always run this to get a different version:

```console
projen new jsii --jsii-version "5.1.x"
```

**Comment by mrgrain** [comment]
Investigated. The issue is probably less the tag itself. But that in order to use a doctag, one has to override the interface property, which is not allowed in jsii.

```ts
export interface HelloOptions extends TypeScriptProjectOptions {
  /**
   * Use TypeScript for your projenrc file (`.projenrc.ts`).
   *
   * @default false
   * @pjnew false
   */
  readonly projenrcTs?: boolean;
}
```

results in:

```
src/index.ts:3:1 - error JSII5015: Interface "pjnew.HelloOptions" re-declares member "projenrcTs". This is not supported as it results in invalid C#.

  3 export interface HelloOptions extends TypeScriptProjectOptions {
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  4   /**
    ~~~~~
... 
 10   readonly projenrcTs?: boolean;
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 11 }
```

**Comment by mrgrain** [comment]
In terms of possible options, I guess we might be looking at introducing a well-known static method or property on a project class, that we can hook into to return a map of the options that should be set when running `projen new`.

**Comment by duckpuppy** [comment]
This is exactly the problem I was describing originally. It's absolutely
that the value in @pjnew can't be overridden. We are trying to develop an
project type that extends the core jsii project type but we need to use an
older jsii  version as default. We could document that creating a new
project requires a user to always add the command line option or edit the
.projenrc.ts file immediately, but being able to handle that internally to
our new project type would be more user friendly.

**Comment by mrgrain** [comment]
Yes, I just needed to write this down in my own words for future reference. 😅

---

Until a feature like this becomes available, your best option is probably [jsii-struct-builder](https://github.com/mrgrain/jsii-struct-builder).
